make[2]: *** No rule to make target 'dist/guide/html/index.html', needed by 'install-data-local'.  Stop.

Cockpit version: Pulled from main branch on July 7, 2021
OS: [Mariner](https://github.com/microsoft/CBL-Mariner)
Page: None

I am unable to compile the newest version of main branch. I get the following error when I do the steps:
1. cd cockpit
2. git pull
3. ./autogen.sh --sysconfdir=/etc --prefix=/usr --enable-debug --disable-pcp --disable-doc
4. make
5. sudo make install
6. See the error:
```
/usr/share/cockpit/shell/index.css.map
/usr/share/cockpit/shell/index.js.map
make[2]: *** No rule to make target 'dist/guide/html/index.html', needed by 'install-data-local'.  Stop.
make[2]: Leaving directory '/root/cockpit'
make[1]: *** [Makefile:9221: install-am] Error 2
make[1]: Leaving directory '/root/cockpit'
make: *** [Makefile:9215: install] Error 2
```

However, I am able to compile the version of main branch on June 14. Here are the steps:
1. cd cockpit
2. git pull
3. git checkout bcd501d1bd8ff602c30fe26cb032b83344cc977f
4. ./autogen.sh --sysconfdir=/etc --prefix=/usr --enable-debug --disable-pcp --disable-doc
5. make
6. sudo make install
7. cockpit is built and installed successfully

Any help would be appreciated to help solve this problem! Thanks!

--- a/Makefile.am
+++ b/Makefile.am
@@ -174,8 +174,13 @@ recheck-memory: valgrind-suppressions
 	        HTML_LOG_FLAGS="valgrind $(VALGRIND_ARGS)" \
 		$(AM_MAKEFLAGS) recheck
 
+if ENABLE_DOC
+DOCS_INSTALL_DEPS = dist/guide/html/index.html
+else
+DOCS_INSTALL_DEPS =
+endif
 
-install-data-local:: dist/guide/html/index.html
+install-data-local:: $(DOCS_INSTALL_DEPS)
 if ENABLE_DOC
 	$(MKDIR_P) $(DESTDIR)$(htmldir)
 	$(INSTALL_DATA) $(dir $<)/* $(DESTDIR)$(htmldir)

